### PR TITLE
Terraform improvements for production

### PR DESCRIPTION
**Pull request by AndrewDryga**

### CODE REVIEW

File: terraform/modules/gateway-google-cloud-compute/templates/cloud-init.yaml [hunk 1]
```diff
@@ -157,6 +157,7 @@ runcmd:
   - sudo apt install docker-ce docker-ce-cli containerd.io -y
   - sudo usermod -aG docker $(whoami)
   - sudo systemctl enable docker
+  - sudo echo '{"ipv6":true,"fixed-cidr-v6":"fd00::/80"}' > /etc/docker/daemon.json
```
Comment (jamilbk) [inline terraform/modules/gateway-google-cloud-compute/templates/cloud-init.yaml:160]: Was this supposed to be within our unique local prefix or outside of it?

**Review by jamilbk — APPROVED** [review]
LGTM, just have one question

File: terraform/modules/gateway-google-cloud-compute/templates/cloud-init.yaml (hunk 1 above)
Comment (AndrewDryga) [reply terraform/modules/gateway-google-cloud-compute/templates/cloud-init.yaml:160]: @jamilbk This is straight what docker tells us to do and I think @bmanifold did the same for AWS gateways

Comment (bmanifold) [reply terraform/modules/gateway-google-cloud-compute/templates/cloud-init.yaml:160]: Yeah, I did follow the instructions on the [Docker IPv6 page](https://docs.docker.com/config/daemon/ipv6/)

The only difference is that I didn't enable it for the default bridge network, but rather created a new network in the docker-compose file.  For reference, here's what I did:

In the `/etc/docker/daemon.json` file
```
{
  "experimental": true,
  "ip6tables": true
}
```

Then in the `docker-compose.yml` file that the gateway is run from:
```


version: '3.8'

services:
  fz_gateway:
    image: us-east1-docker.pkg.dev/firezone-staging/firezone/gateway:1.20231001.0
    ...more-config-here...
    networks:
      - ip6net

networks:
  ip6net:
    enable_ipv6: true
    ipam:
      config:
        - subnet: 2001:0DB8::/112
```

**Pull request merged**